File: beecrowd/1393.c
#include <stdio.h>

int buffer[40] = {0};

int ways(int n) {
    return buffer[n-1] != 0 ? buffer[n-1] : (buffer[n-1] = n == 1 || n == 2 ? n : ways(n-1) + ways(n-2));
}

int main() {
    int n;
    while(1){
        scanf("%d", &n);
        if(n == 0) break;
        printf("%d\n", ways(n));
    }
    return 0;
}

File: beecrowd/1393.py
buffer = [0]*100

def ways(n):
    if a := buffer[n-1]:
        return a
    a = n if n in [1, 2, 3] else ways(n-1) + ways(n-2)
    buffer[n-1] = a
    return a

while(n := int(input())):
    print(ways(n))